q2cli/__main__.py

```diff
@@ -0,0 +1,24 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import click
+
+import q2cli.commands
+import q2cli.info
+
+
+# Entry point for CLI
+@click.command(cls=q2cli.commands.RootCommand, invoke_without_command=True,
+               no_args_is_help=True)
+@click.option('--version', is_flag=True, callback=q2cli.info.echo_version,
+              help='Print the version and exit.', expose_value=False)
+def cli():
+    pass
+
+if __name__ is '__main__':
+    cli()
```

Comment:
The reason for moving the entry-point into this file is we can call:

```bash
python -m q2cli info
```
Which might be useful if a user's environment is totally screwed up.

A `__main__` file is also a pretty obvious place to start looking if you are trying to understand a codebase, so it also felt more natural.